packages/rest/test/integration/http-handler.integration.ts

```diff
@@ -324,14 +346,45 @@ describe('HttpHandler', () => {
       const body = {key: givenLargeRequest()};
       rootContext
         .bind(RestBindings.REQUEST_BODY_PARSER_OPTIONS)
-        .to({limit: 4 * 1024 * 1024}); // Set limit to 4MB
+        .to({limit: '4mb'}); // Set limit to 4MB
       return client
         .post('/show-body')
         .set('content-type', 'application/json')
         .send(body)
         .expect(200, body);
     });
 
+    it('allows request body parser extensions', () => {
```

Comment:
Since most of the body parsing behavior was extracted from HttpHandler into a new BodyParser class, you should also move the relevant tests from `test/integration/http-handler.integration.ts` to a new test file specific for the body-parser implementation. 

It makes sense to keep few body-parsing tests in the integration test suite for HttpHandler to ensure that all parts fit together well, but most of the tests should be at body-parser level.

I think it may be possible to test a lot of body-parsing functionality at unit-test level now, that should simplify the test code a lot.

Paraphrasing https://loopback.io/doc/en/lb4/Defining-your-testing-strategy.html:

To create a great test suite, think smaller and favor fast, focused unit-tests over slow application-wide end-to-end tests.

- One “acceptance test”, where you start the application, make an HTTP request to execute the tested use case and verify the expected HTTP response was returned. This verifies that all parts of the implementation are correctly wired together. (You will probably need more than on acceptance-level test for a feature this big, but I hope you see the point.)

- Few “integration tests” to test integration between a subset of components involved. E.g. BodyParser + individual parser implementations.

- Many “unit tests” where you test individual classes in isolation and verify they handle all different situations, including error paths and edge cases.